astropy/wcs/utils.py

```diff
@@ -71,6 +73,37 @@ def solar_system_body_representation_type(
     )
 
 
+def north_pole_angle(pixel, wcs, ddec=0.1 * u.arcsec):
+    """
+    Computes counterclockwise angle between positive x-axis and sky North
+
+    Parameters
+    ----------
+    pixel : tuple, required.
+        reference pixel in image from which to find angle
+    wcs : `~astropy.wcs.WCS`
+    ddec : `~astropy.coordinates.Angle`, optional, default 0.1 arcseconds.
+        small angular offset for computing direction of north
+        no need to change unless very close to north pole
+
+    Returns
+    -------
+    `~astropy.coordinates.Angle`
+        angle between sky North and coord,
+        along tangent line of great circle running through
+        pixel and sky north.
+    """
+    coord = SkyCoord.from_pixel(pixel[0], pixel[1], wcs)
+    coord = coord.transform_to("icrs")
```

Comment:
I'll point out that we have a number of coordinate frames that which will raise an error when attempting to transform a 2D coordinate from that frame to `ICRS`, due to the ambiguity in how to handle the shift in origin.

Reply:
In that case, maybe the docstring should make it clear that not only it only works with 2D WCS but only with frames compatible with ICRS.

Reply:
We should consider using the default frame for the WCS rather than ICRS which would solve this problem, and then also allow users to override which frame to use via a kwarg as indicated in my comment above.